Civi/Civioffice/Api4/Action/LiveSnippet/GetFieldsAction.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/*
+ * Copyright (C) 2024 SYSTOPIA GmbH
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation in version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types = 1);
+
+namespace Civi\Civioffice\Api4\Action\LiveSnippet;
+
+use Civi\Api4\Generic\BasicGetFieldsAction;
+use Civi\Api4\OptionValue;
+use CRM_Civioffice_ExtensionUtil as E;
+
+final class GetFieldsAction extends BasicGetFieldsAction {
+
+  public function __construct() {
+    parent::__construct('CiviofficeLiveSnippet', 'getFields');
+  }
+
+  /**
+   * @phpstan-return list<array<string, array<string, scalar>|array<scalar>|scalar|null>>
+   */
+  protected function getRecords(): array {
+    $action = OptionValue::getFields($this->getCheckPermissions())
+      ->setAction($this->getAction())
+      ->setLoadOptions($this->getLoadOptions())
+      ->setOrderBy($this->getOrderBy())
+      ->addWhere('name', 'IN', ['label', 'name', 'description'])
+      ->addValue('option_group_id:name', 'civioffice_live_snippets');
+
+    // Property "language" is not available in CiviCRM <5.45
+    // @phpstan-ignore-next-line
+    if (property_exists($this, 'language')) {
+      $action->setLanguage($this->language);
+    }
```

Comment:
I'd say it's ok to raise Core requirement from 5.44 to 5.45.

Reply:
Actually this should be 5.54, not 5.45 https://github.com/civicrm/civicrm-core/commit/cd8cbeb216a516577d231ef46e407ef0d5e32b42

Is it ok to raise to 5.54 or shall I fix the comments?

Reply:
5.54 seems old enough for me to require it.

Reply:
Changed.